pyanaconda/modules/payloads/source/url/url_interface.py

```diff
@@ -0,0 +1,105 @@
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from dasbus.server.interface import dbus_interface
+from dasbus.server.property import emits_properties_changed
+from dasbus.typing import *  # pylint: disable=wildcard-import
+
+from pyanaconda.modules.common.constants.interfaces import PAYLOAD_SOURCE_URL
+from pyanaconda.modules.common.structures.payload import URLData
+from pyanaconda.modules.payloads.source.source_base_interface import PayloadSourceBaseInterface
+
+
+@dbus_interface(PAYLOAD_SOURCE_URL.interface_name)
+class URLSourceInterface(PayloadSourceBaseInterface):
+    """Interface for the payload URL source."""
+
+    def connect_signals(self):
+        super().connect_signals()
+        self.watch_property("URL", self.implementation.url_changed)
+        self.watch_property("SSLVerificationEnabled",
+                            self.implementation.ssl_verification_enabled_changed)
+        self.watch_property("SSLConfiguration", self.implementation.ssl_configuration_changed)
+        self.watch_property("Proxy", self.implementation.proxy_changed)
+
+    @property
+    def URL(self) -> Structure:
+        """Get URL data structure.
+
+        :return: URL data structure containing URL link and type
+        :rtype: URLData data structure
+        """
+        return URLData.to_structure(self.implementation.url)
+
+    @emits_properties_changed
+    def SetURL(self, url: Structure):
+        """Set URL data structure.
+
+        :param url: URL data structure containing URL link and type
+        :type url: URLData data structure
+        """
+        self.implementation.set_url(
+            URLData.from_structure(url)
+        )
+
+    @property
+    def SSLVerificationEnabled(self) -> Bool:
+        """Is ssl verification enabled?"""
+        return self.implementation.ssl_verification_enabled
+
+    @emits_properties_changed
+    def SetSSLVerificationEnabled(self, ssl_verification: Bool):
+        """Enable or disable ssl verification."""
+        self.implementation.set_ssl_verification_enabled(ssl_verification)
+
+    @property
+    def SSLConfiguration(self) -> Structure:
+        """Get custom SSL configuration for this repository.
+
+        :return: SSL configuration
+        :rtype: SSLConfigurationData data structure
+        """
+        return self.implementation.ssl_configuration
+
+    @emits_properties_changed
+    def SetSSLConfiguration(self, ssl_configuration: Structure):
+        """Set custom SSL configuration for this repository.
+
+        :param ssl_configuration: ssl configuration data structure
+        :type ssl_configuration: SSLConfigurationData data structure
+        """
+        self.implementation.set_ssl_configuration(ssl_configuration)
```

Comment:
You should use `SSLConfigurationData` to transform the structure into a data object. Same above.

Reply:
Hmm, true. I just wonder why this wasn't caught by tests.